### ISSUE TITLE
Two "homes" in one home

### ISSUE THREAD

**Issue by rrman** [jvmahon/Homebridge-HomeSeer4#50]
I have about 150 devices that are controlled via homebridge. I use multiple homebridge instances.
Well, my kids just got new iPhones. I'd like them to be able to control some of my home's devices via HomeKit. However, I don't want them to have access to all my home's devices (for example, no access to the water valve). I only want to grant them access to a subset of devices. So, I was thinking of adding another homebridge instance that only has the devices that I want them to have access to. I wouldn't add this new homebridge instance to my iPhone's HomeKit (since I don't want duplication), but I would add the new instance to my kids' HomeKit. So, it would appear that there are two homes in my home. The first home has all 150 or so devices, and the second home, used by my kids, would only be smaller subset of the devices. I suppose that HomeSeer and HomeBridge would be sending/receiving  device status for the same subset of devices twice, and there would be two homebridge instances that would be polling for the same set of devices. 

Do you see any issues with this idea?

Regards,

Randy (randman)

**Comment by jvmahon** [comment]
I don't think this is a problem for my plugin to handle. The overhead from interacting with HomeSeer is pretty minor so it should work. Remember that an instance can only pair with a single phone (but once paired, the bridge can then be shared); so I suggest you may want to pair both homebridge instances to your phone (as separate Homes), that way, you could also create automations on your phone that get shared with your kids phones).

**Comment by rrman** [comment]
Okay, thanks. I don't recall if, when a home is shared, whether it's possible to pick and choose which bridges to share, or if all bridges within the home must be shared. If it's possible to pick only a subset of bridges to share, then I'll only need one home. But if all bridges need to be shared, then I'll have to create a second home with only the devices that I want to share.  I'll have to experiment with this. 

Thus far, I use HomeSeer for all my automations, so some sharing can still be possible even with two homes.

Regards,

Randy

**Comment by jvmahon** [comment]
The entire home is shared.  I agree with using HomeSeer for the automations except those that are geolocation based (for those, I sometimes use the iOS Home automation to trigger a HomeSeer virtual switch which, in turn, triggers the HomeSeer automation.)

**Comment by rrman** [comment]
Yes, geolocation is one of the reasons I want to do this. I've been use GeoFency along with the PHLocation for years. However, with the recent change to iOS 13, iOS now prompts randomly if GeoFency should be allowed to always access location data. I tell people at home to always answer yes, but occasionally they inadvertently answer no, which messes up my location-based events. However, I think if GeoFency is used within HomeKit, iOS doesn't keep randomly prompting users if it should be allowed to continue having location access.

**Comment by jvmahon** [comment]
As further guidance, see this Wiki Entry: https://github.com/jvmahon/Homebridge-HomeSeer4/wiki/Using-GeoFencing-in-HomeSeer

**Comment by rrman** [comment]
Thanks!

**Comment by rrman** [comment]
Hi,

Just wanted to conclude that I created a new homebridge instance, which I added to a second home. This instance only contains a very small subset of devices in my first home. It also includes geofencing devices as you recommended in that link. I then shared this second home with other family members. This second home works great. The geofencing method is a lot simpler than using Geofency and PHLocation (which was my way of doing geofencing that I setup years ago, before I started using your homebridge plugin). Hope this turns out to be reliable in the long run. Probably not as tunable in terms of entry & exit time criteria, but it hopefully is reliable enough. Certainly much simpler and I don't have to worry about iOS randomly prompting users whether to continue to allow the Geofency app permission to use location information (something new in iOS 13).

Again, thanks for your help!
